Name: email-multi
Version: 1.3.5.1-1
Author: cwgtex
Description: Email multi patch.  Combines confirm delete, enable landscape, and swap reply2all with move.  All credit to jhoff80, SirWill, dBsooner, and Jason Robitaille for the original patches.

Index: /usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/accounts-assistant.js
@@ -57,6 +57,8 @@
 		this.boundKeypressHandler = this.keypressHandler.bind(this);
 		this.controller.listen(this.controller.sceneElement, Mojo.Event.keypress, this.boundKeypressHandler);
 		
+		this.controller.window.PalmSystem.setWindowOrientation("free");
+		
 		//set up account folder's search
 		this.folderFilterField = this.controller.get('account_folders_filter');
 		this.boundHandleFoldersSearch = this.handleFoldersSearch.bind(this);
@@ -138,8 +140,8 @@
 	keypressHandler: function(e) {
 		this.orientationString += String.fromCharCode(e.originalEvent.charCode);
 		Mojo.Log.info(this.orientationString);
-		if (this.orientationString.length === 12) {
-			if (this.orientationString === "RocknRollHax") {
+		if (this.orientationString.length === 4) {
+			if (this.orientationString === "wide") {
 				Mojo.Log.info("Let's rock and roll");
 				var targetWindow = this.controller.window;
 				if (targetWindow.PalmSystem.setWindowOrientation) {

Index: /usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/compose-assistant.js
@@ -233,6 +233,8 @@
 		
 		// Delayed a little, since we want the header part of render as quickly as possible
 		ComposeAssistant.onLoad.defer(this.controller, this.email);
+		
+		this.controller.window.PalmSystem.setWindowOrientation("free");
 	},
 
 	cleanup: function() {

Index: /usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/list-assistant.js
@@ -50,6 +50,7 @@
 		visible:true};
 		
 	this.controller.setupWidget(Mojo.Menu.appMenu, undefined, this.appMenuModel);
+	this.controller.window.PalmSystem.setWindowOrientation("free");
   },
 
   // NOTE: this is called by app_scene's _setup function
@@ -66,7 +67,7 @@
 					fixedHeightItems:true,
 					itemTemplate: 'list/email_entry',
 					swipeToDelete: true,
-					autoconfirmDelete: true,
+					autoconfirmDelete: false,
 					uniquenessProperty: 'id',
 					dividerTemplate: 'list/date_separator',
 					dividerFunction: function(email) { return email.dateCategory; },

Index: /usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
===================================================================
--- .orig/usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
+++ /usr/palm/applications/com.palm.app.email/app/controllers/message-assistant.js
@@ -72,6 +72,7 @@
 		this.cmdMenuModel = {
 				visible:true,
 				items: [
+					{label:$L('Move'),      icon:'file',      command:'move'},
 					{label:$L('Reply'),     icon:'reply',     command:'reply'},
 					{label:$L('Reply all'), icon:'reply-all', command:'replyAll'},
 					{label:$L('Forward'),   icon:'forward-email', command:'forward'},
@@ -73,7 +74,6 @@
 				visible:true,
 				items: [
 					{label:$L('Reply'),     icon:'reply',     command:'reply'},
-					{label:$L('Reply all'), icon:'reply-all', command:'replyAll'},
 					{label:$L('Forward'),   icon:'forward-email', command:'forward'},
 					{label:$L('Delete'),    icon:'delete',    command:'delete'}
 				]};
@@ -106,7 +106,7 @@
 				emailEditItems,
 				this.markUnreadMenuItem,
 				this.markSetFlagMenuItem,
-				{label:$L('Move to Folder...'), shortcut:'m', command:'move'},
+				{label:$L('Reply all'), command:'replyAll'},
 				this.showRecipientsMenuItem,
 				Mojo.Menu.prefsItem,
 				Mojo.Menu.helpItem
